dask/dataframe/tests/test_dataframe.py

```diff
@@ -485,3 +485,43 @@ def test_repartition_on_pandas_dataframe():
     assert isinstance(ddf, dd.Series)
     assert ddf.divisions == (10, 20, 50, 60)
     assert eq(ddf, df.y)
+
+
+def test_embarrassingly_parallel_operations():
+    df = pd.DataFrame({'x': [1, 2, 3, 4, None, 6], 'y': list('abdabd')},
+                      index=[10, 20, 30, 40, 50, 60])
+    a = dd.from_pandas(df, 2)
+
+    assert eq(a.x.astype('float32'), df.x.astype('float32'))
+    assert a.x.astype('float32').compute().dtype == 'float32'
+
+    assert eq(a.x.dropna(), df.x.dropna())
+
+    assert eq(a.x.fillna(100), df.x.fillna(100))
+    assert eq(a.fillna(100), df.fillna(100))
+
+    assert eq(a.x.between(2, 4), df.x.between(2, 4))
+
+    assert eq(a.x.clip(2, 4), df.x.clip(2, 4))
+
+    assert eq(a.x.notnull(), df.x.notnull())
+
+    assert len(a.sample(0.5).compute()) < len(df)
+
+
+def test_datetime():
+    df = pd.DataFrame({'x': [1, 2, 3, 4]})
+    df['x'] = df.x.astype('M8[us]')
+
+    a = dd.from_pandas(df, 2)
+
+    assert eq(a.x.dt.date, df.x.dt.date)
+    assert (a.x.dt.to_pydatetime().compute() == df.x.dt.to_pydatetime()).all()
+
+
+def test_datetime():
```

Comment:
Too much copy-pasting going on on that one.

On Thu, Jun 25, 2015 at 2:02 PM, Phillip Cloud <notifications@github.com>
wrote:

> In dask/dataframe/tests/test_dataframe.py
> <https://github.com/ContinuumIO/dask/pull/365#discussion_r33304561>:
>
> > +    assert eq(a.x.notnull(), df.x.notnull())
> > +
> > +    assert len(a.sample(0.5).compute()) < len(df)
> > +
> > +
> > +def test_datetime():
> > +    df = pd.DataFrame({'x': [1, 2, 3, 4]})
> > +    df['x'] = df.x.astype('M8[us]')
> > +
> > +    a = dd.from_pandas(df, 2)
> > +
> > +    assert eq(a.x.dt.date, df.x.dt.date)
> > +    assert (a.x.dt.to_pydatetime().compute() == df.x.dt.to_pydatetime()).all()
> > +
> > +
> > +def test_datetime():
>
> this should be called test_str or something similar to be different from
> the above test_datetime
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/dask/pull/365/files#r33304561>.
>